File: cc/hello_proto.cc
#include <iostream>

#include "google/protobuf/any.pb.h"
#include "protos/my_proto.pb.h"

int main() {
    is_protoc_precompiled::protos::MyMessage msg;
    google::protobuf::Any any;
    any.set_type_url("type.example.com/test");
    any.set_value("hello");
    *msg.mutable_content() = any;
    std::cout << msg.DebugString() << std::endl;
    return 0;
}

File: java/HelloProto.java
import com.google.protobuf.Any;
import com.google.protobuf.ByteString;
import is_protoc_precompiled.protos.MyProto.MyMessage;
import org.junit.Test;

public class HelloProto {
    @Test
    public void buildMessage() {
        Any any = Any.newBuilder()
            .setTypeUrl("type.example.com/test")
            .setValue(ByteString.copyFromUtf8("hello"))
            .build();
        MyMessage msg = MyMessage.newBuilder()
            .setContent(any)
            .build();
        System.out.println(msg);
    }
}
